### ISSUE TITLE
Testing game, engine and tools compilation with GCC10

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1203]
GCC10 is known to break compilation of many idtech-related stuff:

- SpiKe reported me compilation issues on q3map2 (and dæmonmap) I [fixed](https://gitlab.com/xonotic/netradiant/-/issues/140);
- ElBarto reported compilation issues with QStat (and noticed old version of XQF suffered from same issues but git master seems to be OK);
- ET:Legacy has a [work-in-porgress PR](https://github.com/etlegacy/etlegacy/pull/1482) to fix GCC10 issues that are the sames as the one I faced with q3map2.

So unless proven wrong, we have to consider some stuff may break with GCC10 in our game or engine or tools. This was true for dæmonmap (already fixed).

Basically it looks like issues comes from bad coding practices that I'm surprised compilers even did not raised warnings for them, like having a global variable defined with a given type in a `.h` file while a global variable to a `.c` file to produce the same binary defines another global variable for another purpose and using another type with the same name (I would have expected an error since the early days of C), or just redefining the same global variable both in `.h` and `.c` (less problematic but I would have expected a warning).

**Comment by DolceTriade** [comment]
fwiw, I've been using GCC10 to do testing of the stuff for the RmlUi branch.

**Comment by slipher** [comment]
MSYS2 (mingw) has been on GCC 10 for a while. I had to fix a few compiler flag-related issues a while back. Other than that it seems fine.
